### DISCUSSION TITLE
Persistent task should be able to depend on other persistant tasks.

### DISCUSSION THREAD

**Discussion by TheKnarf** [vercel/turborepo#8487] [Ideas]
### Goals

Currently the documentation states the following:

>Because a long-running process won't exit, tasks that would depend on it would never run. Once you've labeled the task as persistent, turbo will throw an error if other tasks depend on it.
>
>This option is most useful for development servers or other "watch" tasks.

I think long-running process should be able to depend on other long-running process.
If your starting a task, which itself is marked as persistent task, and it depends on another task that is also marked as persistent then it should simply just start both tasks in parallell.

I originally laid out my [argument in this issue](https://github.com/vercel/turbo/issues/8484) as I felt the current logic was so unlogical as to be a bug.

### Background

Using [mprocs](https://github.com/pvolok/mprocs) I could run two persistent tasks at the same time:

```bash
pnpm mprocs 'pnpm --filter package-a dev' 'pnpm --filter package-b dev'
```

With turborepo's new persistent mode it should be possible to simply run (assuming that `package-b` depends on `package-a`):

```bash
pnpm turbo watch --filter package-b dev
```

Without this functionality persistent mode makes zero sense. The whole reason people use turborepo in the first place is because they have a monorepo, if tasks can't depend on other tasks then there is no reason to use turborepo.

`pnpm` and `mprocs` already lets me solve this, but `mprocs` have no understanding of the dependency tree like `turbo` does. So if the persistent-model in turbo where to make sense then it should work with dependencies.

### Proposal

Simply remove the warning and let persistent tasks depend on other persistent tasks (thereby running them in parallell).

|Should _column_ be able to depend on _row_|Persistent task|Normal task|
|--|--|--|
|Persisten task| *Yes*, run both in parallell | No, the persistent task will never finish before the normal task gets to run
|Normal task| Yes, run the normal task first and then the persisten task | Yes, run the normal task first before the other normal task

**Comment by anthonyshew** [comment]
Continuing answers on original issue, https://github.com/vercel/turbo/issues/8484
